plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,271 @@
+package elasticsearch
+
+import (
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"gopkg.in/olivere/elastic.v3"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Elasticsearch struct {
+	ServerHost       string
+	IndexName        string
+	EnableSniffer    bool
+	HealthCheck      bool
+	NumberOfShards   int
+	NumberOfReplicas int
+	Client           *elastic.Client
+	Version          string
+}
+
+var sampleConfig = `
+  server_host = "http://10.10.10.10:9200" # required.
+  index_name = "test" # required.
+  # regex allowed on index_name: 
+  # %Y - year (2016)
+  # %y - last two digits of year (00..99)
+  # %m - month (01..12)
+  # %d - day of month (e.g., 01)
+  # %H - hour (00..23)
+  enable_sniffer = false
+  health_check = false
+  number_of_shards = 1
+  number_of_replicas = 0
+`
+
+func (a *Elasticsearch) Connect() error {
+	if a.ServerHost == "" || a.IndexName == "" {
+		return fmt.Errorf("FAILED server_host and index_name are required fields for Elasticsearch output")
+	}
+
+	// Check if index's name has a prefix
+	if strings.HasPrefix(a.IndexName, "%") {
+		return fmt.Errorf("FAILED  Elasticsearch index's name must start with a prefix. \n")
+	}
+
+	client, err := elastic.NewClient(
+		elastic.SetHealthcheck(a.HealthCheck),
+		elastic.SetSniff(a.EnableSniffer),
+		elastic.SetHealthcheckInterval(30*time.Second),
+		elastic.SetURL(a.ServerHost),
+	)
+
+	if err != nil {
+		return fmt.Errorf("FAILED to connect to Elasticsearch host %s : %s\n", a.ServerHost, err)
+	}
+
+	a.Client = client
+
+	version, errVersion := a.Client.ElasticsearchVersion(a.ServerHost)
+
+	if errVersion != nil {
+		return fmt.Errorf("FAILED to get Elasticsearch version : %s\n", errVersion)
+	}
+
+	a.Version = version
+
+	templateName := a.IndexName
+
+	if strings.Contains(a.IndexName, "%") {
+		// Template's name its Index's name without date patterns
+		templateName = a.IndexName[0:strings.Index(a.IndexName, "%")]
+
+		year := strconv.Itoa(time.Now().Year())
+		a.IndexName = strings.Replace(a.IndexName, "%Y", year, -1)
+		a.IndexName = strings.Replace(a.IndexName, "%y", year[len(year)-2:], -1)
+		a.IndexName = strings.Replace(a.IndexName, "%m", fmt.Sprintf("%0.2d", time.Now().Month()), -1)
+		a.IndexName = strings.Replace(a.IndexName, "%d", fmt.Sprintf("%0.2d", time.Now().Day()), -1)
+		a.IndexName = strings.Replace(a.IndexName, "%H", fmt.Sprintf("%0.2d", time.Now().Hour()), -1)
+	}
+
+	exists, errExists := a.Client.IndexExists(a.IndexName).Do()
+
+	if errExists != nil {
+		return fmt.Errorf("FAILED to check if Elasticsearch index %s exists : %s\n", a.IndexName, errExists)
+	}
+
+	if !exists {
```

Comment:
Do we need to do this? could we instead just require that users create an index before using this plugin?